src/main/java/com/redhat/cloud/notifications/events/EventConsumer.java

```diff
@@ -42,38 +42,32 @@ public EventConsumer(MeterRegistry registry) {
     @Acknowledgment(Strategy.PRE_PROCESSING)
     // Can be modified to use Multi<Message<String>> input also for more concurrency
     public Uni<Void> processAsync(Message<String> input) {
-        if (log.isLoggable(Level.FINE)) {
-            log.fine("Processing: " + input.getPayload());
-        }
-        return Uni.createFrom()
-                .item(input)
+        return Uni.createFrom().item(() -> input.getPayload())
+                .onItem().invoke(payload -> log.fine(() -> "Processing: " + payload))
                 .stage(self -> self
                                 // First pipeline stage - modify from Kafka message to processable entity
-                                .onItem().transform(Message::getPayload)
                                 .onItem().transform(this::extractPayload)
                                 .onFailure().invoke(t -> rejectedCount.increment())
-                        // TODO Handle here and set the counters for broken input data and produce empty message?
                 )
                 .stage(self -> self
                                 // Second pipeline stage - enrich from input to destination (webhook) processor format
                                 .onItem()
-                                .transformToUni(action -> destinations.process(action))
-                                .onFailure().invoke(t -> processingErrorCount.increment())
+                                .transformToUni(action -> destinations.process(action)
+                                        .onFailure().invoke(t -> processingErrorCount.increment())
```

Comment:
There's an app behavior change here:
- Before: `processingErrorCount` was incremented when the payload extraction failed at the previous stage, which doesn't make sense.
- Now: it's no longer incremented.

Reply:
> * Before: `processingErrorCount` was incremented when the payload extraction failed at the previous stage, which doesn't make sense.

Could you elaborate on why it doesn't make sense?

Reply:
My understanding of `processingErrorCount` is that it was added to count the failures which happen during a `destinations.process(action)` execution.

With the current version of the code, if `this::extractPayload` fails, `destinations.process(action)` will not be executed and `processingErrorCount` will still be incremented.

Reply:
Ah, understood.

Reply:
Exactly.